File: E_Commerce/E_Commerce/E_Commerce.Dal/Entities/Product.cs
using System.Collections.Generic;

namespace E_Commerce.Dal.Entities;

public class Product
{
    public long ProductId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public int StockQuantity { get; set; }
    public string? ImageLink { get; set; }
    public bool IsDeleted { get; set; }
    public List<CartProduct> CartProducts { get; set; }
    public List<OrderProduct> OrderProducts { get; set; }
}

File: Client/src/app/services/models/Product.ts

import { CartProductModel } from "./CartProduct";
import { OrderProductModel } from "./OrderProduct";

export class ProductModel {
  productId: number = 0;
  name: string = '';
  description: string = '';
  price: number = 0; // decimal in C# maps to number in TypeScript
  stockQuantity: number = 0;
  imageLink?: string | null; // string? in C# maps to string | null | undefined in TypeScript
  isDeleted: boolean = false;
  cartProducts: CartProductModel[] = [];   // List<CartProduct> maps to CartProduct[]
  orderProducts: OrderProductModel[] = []; // List<OrderProduct> maps to OrderProduct[]
}
